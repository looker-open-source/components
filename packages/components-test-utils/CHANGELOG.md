# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.4](https://github.com/looker-open-source/components/compare/v0.10.3...v0.10.4) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.3](https://github.com/looker-open-source/components/compare/v0.10.2...v0.10.3) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.2](https://github.com/looker-open-source/components/compare/v0.10.1...v0.10.2) (2021-01-22)


### Bug Fixes

* On build properly set package.json "files" key ([#1806](https://github.com/looker-open-source/components/issues/1806)) ([c81c4bb](https://github.com/looker-open-source/components/commit/c81c4bb625c58ede49957aad90ac8d9d7b2c4b79))





## [0.10.1](https://github.com/looker-open-source/components/compare/v0.10.0...v0.10.1) (2021-01-22)


### Bug Fixes

* Properly build ems/cjs/ts for easy deep-imports ([#1804](https://github.com/looker-open-source/components/issues/1804)) ([2a523cd](https://github.com/looker-open-source/components/commit/2a523cd70b079944376c6dc87c18202e05a97b01))





# [0.10.0](https://github.com/looker-open-source/components/compare/v0.9.29...v0.10.0) (2021-01-19)


### Bug Fixes

* Properly configure build for tree-shaking support ([9202b72](https://github.com/looker-open-source/components/commit/9202b72bc5fb99eb3e1af7d9e56f3dc15b1df2a4))





## [0.9.15]

### Changed

- `@looker/components-test-utils` helpers will no longer produce `globalStyles`
