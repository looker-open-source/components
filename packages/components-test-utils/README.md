# Test Utilities

Shared functions used for mocking and testing Looker UI Components.
