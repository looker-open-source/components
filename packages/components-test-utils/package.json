{
  "name": "@looker/components-test-utils",
  "license": "MIT",
  "version": "0.9.12",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/looker-open-source/components",
    "directory": "packages/components-test-utils"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@looker/components-providers": "^0.9.12",
    "@testing-library/react": "^10.4.8",
    "@types/enzyme": "^3.10.5",
    "@types/jest": "^26.0.10",
    "@types/react": "^16.9.49",
    "@types/react-test-renderer": "^16.9.3",
    "enzyme": "^3.11.0",
    "jest-styled-components": "^6.3.4",
    "react": "^16.13.1",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^4.4.1"
  },
  "peerDependencies": {
    "react": "^16.11"
  },
  "gitHead": "62febc162e3e45124f403e2c84fc741f68fe6714"
}
