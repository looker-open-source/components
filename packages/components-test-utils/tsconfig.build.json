{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./lib"
  },
  "include": ["../../types", "src/**/*"]
}
