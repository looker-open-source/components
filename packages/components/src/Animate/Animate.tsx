/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import { CompatibleHTMLProps, fadeIn, Transitions } from '@looker/design-tokens'
import omit from 'lodash/omit'
import React from 'react'
import styled, { css } from 'styled-components'
import { simpleLayoutCSS, SimpleLayoutProps } from '../Layout/utils/simple'

export type AnimationProps = {
  /**
   * Adds a delay before the start of the animation
   * @default none
   */
  delay?: Transitions
  /**
   * Controls the duration of the animation
   * @default quick
   */
  duration?: Transitions
}

export type AnimateProps = AnimationProps &
  SimpleLayoutProps &
  CompatibleHTMLProps<HTMLDivElement>

export const animateCSS = css<AnimationProps>`
  animation-delay: ${({ delay = 'none', theme }) =>
    `${theme.transitions[delay]}ms`};
  animation-duration: ${({ duration = 'quick', theme }) =>
    `${theme.transitions[duration]}ms`};
`

export const Animate = styled((props: AnimateProps) => (
  <div {...omit(props, ['delay', 'duration'])} />
))`
  ${simpleLayoutCSS}
  ${animateCSS}
`

export const FadeIn = styled(Animate)`
  animation-fill-mode: forwards;
  animation-name: ${fadeIn};
  opacity: 0;
`
