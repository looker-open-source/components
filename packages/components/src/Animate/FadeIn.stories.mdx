import { FadeIn } from '@looker/components'
import { Meta } from '@storybook/addon-docs'

<Meta component={FadeIn} title="Docs/Components/Content/FadeIn" />

# FadeIn

`FadeIn` renders an animated container to wrap around content that you want to fade in,
with `duration` and `delay` props to customize the animation.

```jsx
;() => {
  const { value, toggle } = useToggle(false)
  return (
    <>
      <Button onClick={toggle}>Toggle Content</Button>
      {value && (
        <>
          <FadeIn>This fades in.</FadeIn>
          <FadeIn delay="complex" duration="intricate">
            This fades in more slowly.
          </FadeIn>
        </>
      )}
    </>
  )
}
```
