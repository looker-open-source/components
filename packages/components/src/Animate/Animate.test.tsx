/*

 MIT License

 Copyright (c) 2021 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React from 'react'
import { screen } from '@testing-library/react'
import { renderWithTheme } from '@looker/components-test-utils'
import { FadeIn } from './Animate'

describe('FadeIn', () => {
  it('defaults', () => {
    renderWithTheme(<FadeIn data-testid="fadein" />)
    expect(screen.getByTestId('fadein')).toBeInTheDocument()
    expect(screen.getByTestId('fadein')).toHaveStyle(
      'animation-fill-mode: both;'
    )
  })
  it('delay and duration props', () => {
    renderWithTheme(
      <FadeIn data-testid="fadein" delay="intricate" duration="complex" />
    )
    expect(screen.getByTestId('fadein')).toHaveStyle('animation-delay: 500ms;')
    expect(screen.getByTestId('fadein')).toHaveStyle(
      'animation-duration: 400ms;'
    )
  })

  it('elements inside', () => {
    renderWithTheme(
      <FadeIn>
        <span>Some text</span>
      </FadeIn>
    )
    expect(screen.getByText('Some text')).toBeVisible()
  })
})
