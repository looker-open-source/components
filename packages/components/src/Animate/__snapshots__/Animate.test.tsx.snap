// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeIn renders with defaults 1`] = `
.c0 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
}

.c1 {
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: lbWRkT;
  animation-name: lbWRkT;
  opacity: 0;
}

<div
  className="c0 c1"
/>
`;

exports[`FadeIn renders with delay and duration props 1`] = `
.c0 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  -webkit-animation-delay: 500ms;
  animation-delay: 500ms;
  -webkit-animation-duration: 400ms;
  animation-duration: 400ms;
}

.c1 {
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: lbWRkT;
  animation-name: lbWRkT;
  opacity: 0;
}

<div
  className="c0 c1"
/>
`;
